### PR TITLE
[TS] Update UI Test App and add first sample set of tests 

### LINKED ISSUE #5927: [TS] Update index.html file in nodejs project to handle UI Tests

### PR DESCRIPTION
**Pull request by almedina-ms**
# Related Issue

Fixes #5927 

# Description

Modifies the test app to render a list of test cases, the last selected card and the retrieved inputs as a json text. 

* To add more Test Cases just add them to the corresponding list in `adaptivecards-ui-testapp/src/adaptivecards-ui-testapp.ts`, currently all test samples are copied to the app folder but they can't be automatically read on app start up.

Adds sample tests to the ui-tests project to serve as demonstrative code, as usual the test cases were interacting with the ActivityUpdate card filling in data and verifying the result and interacting with the Input.Text.ErrorMessage card and verifying the first element got focus.

# How Verified

This is a test PR, the tests were executed locally

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/6227)

### CODE REVIEW

File: source/nodejs/adaptivecards-ui-testapp/index.html [hunk 1]
```diff
@@ -6,12 +6,71 @@
 
 		<script src="https://unpkg.com/markdown-it@8.4.0/dist/markdown-it.js"></script>
```
Comment (paulcam206) [inline source/nodejs/adaptivecards-ui-testapp/index.html:7]: not part of this review -- feel free to move to a bug:

retrieving `markdown-it` over the wire might not be the best idea in CI. it's probably better to `npm install markdown-it` in this project and get it bundled with webpack

File: source/nodejs/adaptivecards-ui-testapp/src/adaptivecards-ui-testapp.ts [hunk 2]
```diff
@@ -1 +1,39 @@
...
+        fixTestCases(v11TestCases, "1.1"),
+        fixTestCases(v12TestCases, "1.2"),
+        fixTestCases(v13TestCases, "1.3"),
+        fixTestCases(v14TestCases, "1.4"),
+        fixTestCases(v15TestCases, "1.5"));
+
+    return testCasesList;
+}
+
+export {getTestCasesList};
```
Comment (paulcam206) [inline source/nodejs/adaptivecards-ui-testapp/src/adaptivecards-ui-testapp.ts:39]: this should probably be extracted out into its own file... as we add more functionality, we'll want to keep things organized :)

File: source/nodejs/adaptivecards-ui-testapp/src/adaptivecards-ui-testapp.ts [hunk 3]
```diff
@@ -1 +1,39 @@
 export { AdaptiveCard, HostConfig } from "adaptivecards";
```
Comment (paulcam206) [inline source/nodejs/adaptivecards-ui-testapp/src/adaptivecards-ui-testapp.ts:1]: missing copyright banner

File: source/nodejs/tests/ui-tests/src/testUtils.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        this.driver = driver;
+    }
+
+    async goToTestCase(testCaseName: string): Promise<void> {
+        const elementLinkText: Webdriver.WebElement = await this.driver.findElement(Webdriver.By.id(testCaseName));
+        await elementLinkText.click();
+    }
+
+    async getActionWithTitle(actionTitle: string): Promise<Webdriver.WebElement> {
+        const button: Webdriver.WebElement = await this.driver.findElement(Webdriver.By.xpath("//*[@aria-label='" + actionTitle + "']"));
```
Comment (paulcam206) [inline source/nodejs/tests/ui-tests/src/testUtils.ts:18]: might be a little clearer if you use string interpolation:

```suggestion
        const button: Webdriver.WebElement = await this.driver.findElement(Webdriver.By.xpath(`//*[@aria-label='${actionTitle}']`));
```

File: source/nodejs/tests/ui-tests/src/ui-tests.test.ts [hunk 5]
```diff
@@ -2,22 +2,52 @@
...
     afterAll(async function() {
         if (driver) {
-            // driver.quit();
+            await driver.quit();
         }
     });
 
-    test("Test case should work", async function() {
-        Assert.equal(true, true);
+    test("Test ActivityUpdate submit", async function() {
```
Comment (paulcam206) [inline source/nodejs/tests/ui-tests/src/ui-tests.test.ts:24]: nothing wrong with this syntax, but an alternative you might prefer:

```javascript
test("text", (async () => {
    // body of test
}));
```

**Review by paulcam206 — CHANGES_REQUESTED** [review]

File: source/nodejs/adaptivecards-ui-testapp/index.html (hunk 1 above)
Comment (almedina-ms) [reply source/nodejs/adaptivecards-ui-testapp/index.html]: most of the places where this library is used are not executed/run/opened during CI or any kind of build, does it make sense to open the bug for all projects

File: source/nodejs/adaptivecards-ui-testapp/index.html [hunk 6]
```diff
@@ -4,71 +4,33 @@
...
+			elementList.forEach(element => {
+				testCaseListElement.appendChild(element);
+			});
+		}
+
+		function renderingAdaptiveCardFinishedCallback(renderedCard) {
+			var renderedCardDiv = document.getElementById("renderedCardSpace");
+    		renderedCardDiv.innerHTML = "";
+    		renderedCardDiv.appendChild(renderedCard);
+		}
```
Comment (paulcam206) [inline source/nodejs/adaptivecards-ui-testapp/index.html:24]: nit: indenting and/or whitespace

**Review by paulcam206 — APPROVED** [review]
looks good! (only one minor formatting nit)

File: source/nodejs/adaptivecards-ui-testapp/index.html (hunk 1 above)
Comment (paulcam206) [reply source/nodejs/adaptivecards-ui-testapp/index.html]: I wouldn't worry about it, and it's not a problem that really needs to be solved right now :)

File: source/nodejs/adaptivecards-ui-testapp/index.html [hunk 7]
```diff
@@ -4,14 +4,35 @@
         <meta charset="utf-8" />
         <title>Adaptive Cards UI TestApp</title>
```
Comment (RebeccaAnne) [inline source/nodejs/adaptivecards-ui-testapp/index.html:5]: >         [](http://example.com/codeflow?start=0&length=8)

[nit] Tabs vs spaces inconsistency here.

**Review by RebeccaAnne — APPROVED** [review]
:shipit:

**Pull request merged**